frameworks/Dart/aqueduct/server.dart

```diff
@@ -0,0 +1,228 @@
+  DartAqueductBenchmarkSink(ApplicationConfiguration appConfig)
+      : super(appConfig) {
+    // Use logging for debuging only
+//    logger.onRecord.listen(
+//        (rec) => print("$rec ${rec.error ?? ""} ${rec.stackTrace ?? ""}"));
+
+    var options =
+        new DartAqueductBenchmarkConfiguration(appConfig.configurationFilePath);
+    ManagedContext.defaultContext = contextWithConnectionInfo(options.database);
+
+    Future.wait([
+      new File('fortunes.mustache').readAsString().then((template) {
+        fortunesTemplate = new mustache.Template(template);
+      })
+    ]);
+  }
+
+  /// All routes must be configured in this method.
+  ///
+  /// This method is invoked after the constructor and before [willOpen].
+  /// All routes must be set up in this method and cannot be added after this method completes.
+  @override
+  void setupRouter(Router router) {
+    router
+        .route("/json")
+        .listen((req) async => new Response.ok({"message": "Hello, World!"})
+          ..contentType = _stripped_json
+          ..headers["date"] = new DateTime.now());
+
+    router
+        .route("/plaintext")
+        .listen((req) async => new Response.ok("Hello, World!")
+          ..contentType = _stripped_text
+          ..headers["date"] = new DateTime.now());
+
+    router.route("/db").listen((req) async {
+      World result = await getRandomWorldObject();
+      return new Response.ok(result)
+        ..contentType = _stripped_json
+        ..headers["date"] = new DateTime.now();
+    });
+
+    router.route("/queries/[:queryCount]").listen((req) async {
+      int queryCount = _min_query_count;
+      if (req.path.variables.containsKey("queryCount")) {
+        queryCount = int
+            .parse(req.path.variables["queryCount"], onError: (_) => queryCount)
+            .clamp(_min_query_count, _max_query_count);
+      }
+      List<Future> resultFutures =
+          new List.generate(queryCount, (_) => getRandomWorldObject());
+      List results = await Future.wait(resultFutures);
+      return new Response.ok(results)
+        ..contentType = _stripped_json
+        ..headers["date"] = new DateTime.now();
+    });
+
+    router.route("/updates/[:queryCount]").listen((req) async {
+      int queryCount = _min_query_count;
+      if (req.path.variables.containsKey("queryCount")) {
+        queryCount = int
+            .parse(req.path.variables["queryCount"], onError: (_) => queryCount)
+            .clamp(_min_query_count, _max_query_count);
+      }
+      List<Future> resultFutures = new List.generate(
+          queryCount,
+          (_) => getRandomWorldObject().then((World world) async {
+                Query query = new Query<World>()
+                  ..where.id = whereEqualTo(world.id)
+                  ..values.randomNumber =
+                      (_random.nextInt(_world_table_size) + 1);
+                Future<World> result = query.updateOne();
+                return await result;
```

Comment:
no need to await on result here, in fact, it'll be the tiniest of slowdowns. can just `return query.updateOne()`